src/core/Global.ttslua

```diff
@@ -888,26 +885,32 @@ end
 ---@param name String Name of the object that should be copied
 ---@param position Position Desired position of the object
 function spawnHelperObject(name, position, rotation, color)
-  for _, obj in ipairs(getObjectFromGUID(BARREL_GUID).getData().ContainedObjects) do
-    if obj["Nickname"] == name then
-      return spawnObjectData({
-        data = obj,
-        position = position,
-        rotation = rotation or {0, 270, 0},
-        callback_function = function(object)
+  local spawnTable = {
+    position = position,
+    callback_function = function(object)
           if name == "Hand Helper" then
             Wait.time(function() object.call("externalColorChange", color) end, 1)
           elseif name == "Token Arranger" then
             Wait.time(function() object.call("layout") end, 1)
           end
-        end
-      })
+    end
+  }
+
+  -- only overrride rotation if there is one provided (object's rotation used instead)
+  if rotation then
+    spawnTable.rotation = rotation
+  end
+
+  for _, obj in ipairs(getObjectFromGUID(BARREL_GUID).getData().ContainedObjects) do
```

Comment:
Not sure if this is new or I missed it before, but this will throw an error if the barrel happens to be deleted.  Would be better to check for a nil and send a broadcast, then ensure that the option state is accurate (i.e. off).  Can consider for a future enhancement, assuming we move the barrel somewhere hidden it's low-risk.

Reply:
Error handling added:

![image](https://user-images.githubusercontent.com/97286811/209133262-1fdfdfd5-608e-40b3-9b58-fe48f71ae311.png)

Unfortunately I can't get rid of the TTS error:
![image](https://user-images.githubusercontent.com/97286811/209133182-488dd5da-fbdc-41d7-ac05-91749bc2cc51.png)

From my understanding the `return` should stop the rest of the function to execute, but apparently it's still trying to do some of it?